zebra-state/src/response.rs

```diff
@@ -41,3 +44,13 @@ pub enum Response {
     /// The response to a `FindBlockHeaders` request.
     BlockHeaders(Vec<block::CountedHeader>),
 }
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+/// A response to a read-only [`ReadStateService`] [`ReadRequest`].
+pub enum ReadResponse {
```

Comment:
Should this and `ReadRequest` live in `(request|response)::read` and be accessed via `request::read::Request` and `response::read::Response`?

Reply:
I'd prefer not to do that, for a few reasons:

Conceptual:
1. We'll modify both kinds of requests at the same time, so they belong in the same module
2. If someone is looking for a specific state request variant, they can read the whole module to find them all
3. `read` contains shared reading code, not `ReadService`-only code

Ergonomic:
1. It's harder to import both request types in the same file
2. It's easier to confuse the request types
3. Some editors show both types as `Request`, which is confusing